client/sites/components/sites-dataviews/style.scss

```diff
@@ -101,6 +101,35 @@
 	.stats-sparkline {
 		padding-top: 0;
 	}
+
+	/**
+	 * Pin the actions column.
+	 */
+	.dataviews-view-table__actions-column {
```

Comment:
Hey, we shouldn't target CSS classes of libraries (including DataViews), otherwise updates are harder for us to perform.

The way forward here is to use `@automattic/dataviews` and implement these changes there. In case it's useful as a starting point to port here, there was an initial PR for that at https://github.com/WordPress/gutenberg/pull/65086.

Reply:
I've moved changes to `@automattic/dataviews`. Thank you!

Reply:
Do you think I should split this PR and add support for pinning action columns in dataviews separately?

Reply:
> Do you think I should split this PR and add support for pinning action columns in dataviews separately?

Perhaps a good idea.

`@automattic/dataviews` is a 1:1 replacement for `@wordpress/dataviews` 4.18.0, so it should be fine. However, we're starting to do stuff, and changing the package merits some deeper testing given other code bits that we have in dotcom. Worst-case scenario, a revert would be easier if it's just a package switch (without losing the feature).